applets/clock/set-timezone.c

```diff
@@ -171,75 +178,63 @@ free_data (gpointer d)
 }
 
 static void
-set_time_notify (DBusGProxy     *proxy,
-		 DBusGProxyCall *call,
-		 void           *user_data)
+set_time_notify (GObject *source_object,
+		GAsyncResult *res,
+		gpointer user_data)
 {
 	SetTimeCallbackData *data = user_data;
-	GError *error = NULL;
-
-	if (dbus_g_proxy_end_call (proxy, call, &error, G_TYPE_INVALID)) {
-		if (data->callback) 
-			data->callback (data->data, NULL);
-	}
-	else {
-		if (error->domain == DBUS_GERROR &&
-		    error->code == DBUS_GERROR_NO_REPLY) {
-			/* these errors happen because dbus doesn't
-			 * use monotonic clocks
-			 */	
-			g_warning ("ignoring no-reply error when setting time");
-			g_error_free (error);
-			if (data->callback)
-				data->callback (data->data, NULL);
-		}
-		else {
+	GError     *error = NULL;
+	GDBusProxy *proxy;
+	GVariant   *variant;
+
+	proxy = get_bus_proxy ();
+	variant = g_dbus_proxy_call_finish (proxy, res, &error);
+	if (variant == NULL) {
+		if (error != NULL) {
 			if (data->callback)
 				data->callback (data->data, error);
 			else
 				g_error_free (error);
-		}		
+		} else {
+			if (data->callback)
+				data->callback (data->data, NULL);
+		}
+	} else {
+		g_variant_get (variant, "()");
+		g_variant_unref (variant);
+		if (data->callback)
+			data->callback (data->data, NULL);
 	}
 }
 
 static void
 set_time_async (SetTimeCallbackData *data)
 {
-        DBusGConnection *bus;
-        DBusGProxy      *proxy;
+        GDBusProxy      *proxy;
 
-        bus = get_system_bus ();
-        if (bus == NULL)
-                return;
-
-	proxy = dbus_g_proxy_new_for_name (bus,
-					   "org.mate.SettingsDaemon.DateTimeMechanism",
-					   "/",
-					   "org.mate.SettingsDaemon.DateTimeMechanism");
+	proxy = get_bus_proxy ();
+	if (proxy == NULL)
+		return;
 
 	data->ref_count++;
 	if (strcmp (data->call, "SetTime") == 0)
-		dbus_g_proxy_begin_call_with_timeout (proxy, 
-						      "SetTime",
-						      set_time_notify,
-						      data, free_data,
-						      INT_MAX,
-						      /* parameters: */
-						      G_TYPE_INT64, data->time,
-						      G_TYPE_INVALID,
-						      /* return values: */
-						      G_TYPE_INVALID);
+		g_dbus_proxy_call (proxy,
+				"SetTime",
```

Comment:
Does not align correctly...

Reply:
My personal opinion is that tabs should not be used, only spaces. To keep existing style use combination of tab/tabs and spaces. For example:
```
[tab]my_function (param1,
[tab][spaaaaaaces]param2);
```

Reply:
Good. It looks like a better way.
Is there any possibility of using tools?